src/context/WalletProvider/NativeWallet/components/LegacyLogin.tsx

```diff
@@ -110,9 +115,12 @@ export const LegacyLogin = () => {
     }
   }
 
-  const onCaptchaRequested = (e: React.ChangeEvent<HTMLInputElement>) => {
-    // Check the captcha in case the captcha has been validated
-    setIsCaptchaSolved(e.target.checked)
+  const handleCaptcha = (solution: string) => {
+    if (typeof solution === 'string') {
+      setCaptchaSolution(solution)
+    } else {
+      setCaptchaSolution(null)
+    }
   }
```

Comment:
```suggestion
```

Reply:
also - this type sig is different to another component where it's `string | any` - which is it?

Reply:
if we just pass down `setCaptchaSolution` to the child it'll be referentially stable and we don't need the callbacks in the child either?